### PR TITLE
Roger helm

### PR DESCRIPTION
**Pull request by YaphetKG**
For the values in running this chart please refer to (https://github.com/helxplatform/roger/blob/consolidated-helm/helm/roger/Readme.md) as the readme is not included here.

Note the pre-install steps in the read me (the PVC creation and secrets) .

And last thing is set the value of `.Values.airflow.airflow.config.AIRFLOW_BASE_URL` to `http(s):\\{ helx-nginx-host-name }\{ .Values.web.ambassador.airflowUI.prefix }` , this need to be done on the values.yaml file. 

This installation will add the following Urls to ambassador : 
 `/tranql`  --- > tranql 
`/admin/` ---- > airflow admin url
`/static/` -----> airflow static url
`/airflow/ ----> airflow landing page (This will auto redirect to admin)

### CODE REVIEW

**Comment by pj-linebaugh** [comment]
/admin is used for the Django admin website

File: helx/charts/roger/values.yaml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    ##
+    image:
+      repository: renciorg/apache-airflow-1.10.14-python-3.8-git
+      tag: latest
+    #  executor: CeleryExecutor
+    executor: KubernetesExecutor
+
+    ## the fernet key used to encrypt the connections in the database
+    ##
+    fernetKey: "7T512UXSSmBOkpWimFHIVb8jK6lfmSAvx4mO6Arehnc="
```
Comment (pj-linebaugh) [inline helx/charts/roger/values.yaml:52]: What's this used for and should it be here?  Set it to a random value on deployment?

File: helx/charts/roger/Chart.lock [hunk 2]
```diff
@@ -0,0 +1,9 @@
+dependencies:
```
Comment (pj-linebaugh) [inline helx/charts/roger/Chart.lock:1]: Is this a temp file?

File: helx/charts/roger/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+charts/
```
Comment (pj-linebaugh) [inline helx/charts/roger/.gitignore:1]: Don't think this is needed.

**Review by pj-linebaugh — CHANGES_REQUESTED** [review]
Added a few comments in the files.

File: helx/charts/roger/templates/airflow-service-temp.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      service: {{ include "roger.fullname" . }}-airflow-ui-svc:8080
+      rewrite: /static
+      timeout_ms: 6000
+      ---
+      apiVersion: ambassador/v1
+      kind: Mapping
+      name: {{ .Values.web.ambassador.airflowUI.map_name }}-3
+      prefix: /admin
+      service: {{ include "roger.fullname" . }}-airflow-ui-svc:8080
+      rewrite: /admin
```
Comment (pj-linebaugh) [inline helx/charts/roger/templates/airflow-service-temp.yaml:31]: /admin will conflict with Django's /admin path.

**Review by pj-linebaugh — CHANGES_REQUESTED** [review]
/admin confiict

File: helx/charts/roger/Chart.lock (hunk 2 above)
Comment (YaphetKG) [reply helx/charts/roger/Chart.lock:1]: This is lets rebuild the charts/ directory to the state specified in the lock file (https://helm.sh/docs/helm/helm_dependency_build/). This was generated using helm dependency update (https://helm.sh/docs/helm/helm_dependency_update/)  , I think the advantage is more precision on the dependency version (?)

File: helx/charts/roger/.gitignore (hunk 3 above)
Comment (YaphetKG) [reply helx/charts/roger/.gitignore:1]: The charts/ dir  on helm dependency build/update will download zip files for airflow and redis charts , didn't want to add those in in git

File: helx/charts/roger/values.yaml (hunk 1 above)
Comment (YaphetKG) [reply helx/charts/roger/values.yaml]: Fernet key is now auto-generated (https://github.com/helxplatform/devops/blob/roger-helm/helx/charts/roger/templates/airflow-config-secrets.yaml ) and used  [here](https://github.com/helxplatform/devops/blob/bfe61fb840ecb10d27422ca0639a4596f195a20a/helx/charts/roger/values.yaml#L100)

File: helx/charts/roger/templates/airflow-service-temp.yaml (hunk 4 above)
Comment (YaphetKG) [reply helx/charts/roger/templates/airflow-service-temp.yaml]: This takes care of putting every thing under `/airflow` prefix ,(https://github.com/helxplatform/devops/blob/bfe61fb840ecb10d27422ca0639a4596f195a20a/helx/charts/roger/values.yaml(#L74 )

Comment (YaphetKG) [reply helx/charts/roger/templates/airflow-service-temp.yaml]: now path is under `/airflow/admin/....`

Comment (YaphetKG) [reply helx/charts/roger/templates/airflow-service-temp.yaml]: This is also accessible iff user is  authorized through appstore

File: helx/charts/roger/Chart.lock (hunk 2 above)
Comment (pj-linebaugh) [reply helx/charts/roger/Chart.lock:1]: I believe this file will be used to download the tar archives of the dependencies.  I'll need to do a "helm dependency update" when I add this to the devops helm repo.

File: helx/charts/roger/.gitignore (hunk 3 above)
Comment (pj-linebaugh) [reply helx/charts/roger/.gitignore]: At first I was thinking that only the root .gitignore would be used, so I learned something new.  Now I'm thinking this shouldn't be here b/c when I build the charts the zip files will need to be checked in for the devops helm repo.

Comment (YaphetKG) [reply helx/charts/roger/.gitignore]: This makes sense, maybe we could do a rule like ignore all zips except the ones you have built under (`docs/charts/`), at the root .gitignore ? 
Also when you said they shouldn't be here is

Comment (pj-linebaugh) [reply helx/charts/roger/.gitignore]: I was referring to roger/.gitignore.  I believe we can use something like "!*.zip" or "!*.tar.gz" in roger/.gitignore.  I haven't really needed an ignore for .zip or .tar.gz files so far.

**Comment by YaphetKG** [comment]
Closing to PR to reopen for clarity of outdated changes.

**Pull request closed without merging**